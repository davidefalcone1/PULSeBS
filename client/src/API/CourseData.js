//usata per la retrieve della lista dei corsi di un docente
class CourseData{
    constructor(courseId, courseName, teacherId, 
        normalBookingsAvgWeek, cancelledBookingsAvgWeek, waitingBookingsAvgWeek, 
        normalBookingsAvgMonth, cancelledBookingsAvgMonth, waitingBookingsAvgMonth,
        attendanceCountAvgMonth, attendanceCountAvgWeek ) {
        if (courseId) 
            this.courseId = courseId;
        this.courseName = courseName;
        this.teacherId = teacherId;
        this.normalBookingsAvgWeek = normalBookingsAvgWeek;
        this.cancelledBookingsAvgWeek = cancelledBookingsAvgWeek;
        this.waitingBookingsAvgWeek = waitingBookingsAvgWeek;
        this.normalBookingsAvgMonth = normalBookingsAvgMonth;
        this.cancelledBookingsAvgMonth = cancelledBookingsAvgMonth;
        this.waitingBookingsAvgMonth = waitingBookingsAvgMonth;
        this.attendanceCountAvgWeek = attendanceCountAvgWeek;
        this.attendanceCountAvgMonth = attendanceCountAvgMonth;
    }

    static fromJson(json){
        const temp =  Object.assign(new CourseData(), json);
        return temp;
    }
}

export default CourseData;