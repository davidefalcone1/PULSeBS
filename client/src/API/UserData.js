class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

    static fromJson(json){
        const temp =  Object.assign(new UserData(), json);
        return temp;
    }
}

export default UserData;