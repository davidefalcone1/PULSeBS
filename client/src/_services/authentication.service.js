import { BehaviorSubject } from 'rxjs';


const currentUserSubject = new BehaviorSubject(JSON.parse(localStorage.getItem('user')));

export const authenticationService = {
    login,
    logout,
    user: currentUserSubject.asObservable(),
    get currentUserValue () { return currentUserSubject.value }
};

function handleResponse(response) {
    return response.text().then(text => {
        const data = text && JSON.parse(text);
        if (!response.ok) {
            if ([401, 403].indexOf(response.status) !== -1) {
                // auto logout if 401 Unauthorized or 403 Forbidden response returned from api
                authenticationService.logout();
            }

            const error = (data && data.message) || response.statusText;
            return Promise.reject(error);
        }

        return data;
    });
}

function login(username, password) {
    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ username, password })
    };
    return new Promise(async function (resolve, reject) {
        try {
            const response = await fetch('/users/authenticate', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ username, password }),
            });
            const user = await response.json().then(handleResponse);
            if (response.ok) {
                localStorage.setItem('user', JSON.stringify(user));
                currentUserSubject.next(user);
                resolve(user);
            }
            else
                reject();
        }
        catch (e) {
            reject();
        }
    });

}
function logout() {

    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
    };
    localStorage.removeItem('user');
    currentUserSubject.next(null);
    return fetch('/logout',requestOptions);
}