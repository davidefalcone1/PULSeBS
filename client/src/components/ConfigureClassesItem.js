import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';

const configureClassroomItem = (props) => {
  return (
    <ListGroup.Item id = {"class-" + props.enrollment.studentId + '-' + props.enrollment.courseId}>
        <div className="d-flex w-100 pt-3 justify-content-between no-gutters">
            <CourseField id = {"c-" + props.enrollment.courseId + "-" + props.enrollment.studentId}
                courseId = {props.enrollment.courseId} courses = {props.courses}/>
            <StudentField id = {"s-" + props.enrollment.courseId + "-" + props.enrollment.studentId}
                studentId = {props.enrollment.studentId} students = {props.students}/>
        </div>
    </ListGroup.Item>
  );
}

function CourseField(props){
    return(
        <div className="col-sm-6">
            {props.courses.map((c) => //per ogni lezione del mio corso
                (c.courseId === props.courseId) &&
                    <h6 id={"nameOfcourseForEnrollement_" + props.id} key={"nameOfcourseForEnrollement_" + props.id}>
                        {c.courseName}
                    </h6>
            )}
        </div>
    );
}

function StudentField(props){
    return(
        <div className="col-sm-6">
            {props.students.map((s) => //per ogni lezione del mio corso
                (s.personId === props.studentId) &&
                    <h6 id={"nameOfstudentForEnrollement_" + props.id} key={"nameOfstudentForEnrollement_" + props.id}>
                        {s.fullName}
                    </h6>
            )}
        </div>
    );
}

export default configureClassroomItem;