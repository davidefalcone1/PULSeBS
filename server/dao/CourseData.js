//usata per la retrieve della lista dei corsi
class CourseData{
    constructor(courseId, courseName, teacherId) {
        if (courseId) 
            this.courseId = courseId;
        this.courseName = courseName;
        this.teacherId = teacherId;
    }
}

module.exports = CourseData;