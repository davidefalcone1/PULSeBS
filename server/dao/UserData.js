class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

}

module.exports = UserData;