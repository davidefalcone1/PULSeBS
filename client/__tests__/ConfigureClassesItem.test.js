import React from 'react';
import renderer from 'react-test-renderer';
import { AuthContext } from '../src/_services/AuthContext';
import ConfigureClassesItem from '../src/components/ConfigureClassesItem';
import { BrowserRouter } from 'react-router-dom';
import CourseData from '../src/API/CourseData';
import UserData from '../src/API/UserData';

describe('ConfigureClassesItem', ()=>{
    test('Renders correctly', ()=>{
        const component = renderer.create(
            <BrowserRouter>
                <AuthContext.Provider value={{user: undefined}}>
                    <ConfigureClassesItem enrollment={{courseId: 'XY123', studentId: '12345'}} courses={[new CourseData('XY123', 'Software engineering 2', '654321', 2, 3, 5, 6, 7, 3, 5, 7)]} students={[new UserData(1, '12345', 'Davide Falcone', 'davide.falcone@polito.it')]}/>
                </AuthContext.Provider>
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});