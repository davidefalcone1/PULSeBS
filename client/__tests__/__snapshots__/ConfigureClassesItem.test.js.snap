// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureClassesItem Renders correctly 1`] = `
<div
  className="list-group-item"
  disabled={false}
  id="class-12345-XY123"
  onClick={[Function]}
>
  <div
    className="d-flex w-100 pt-3 justify-content-between no-gutters"
  >
    <div
      className="col-sm-6"
    >
      <h6
        id="nameOfcourseForEnrollement_c-XY123-12345"
      >
        Software engineering 2
      </h6>
    </div>
    <div
      className="col-sm-6"
    >
      <h6
        id="nameOfstudentForEnrollement_s-XY123-12345"
      >
        Davide Falcone
      </h6>
    </div>
  </div>
</div>
`;
